src/ai/backend/web/proxy.py

```diff
@@ -399,6 +400,25 @@ async def websocket_handler(request, *, is_anonymous=False) -> web.StreamRespons
                 content_type=request.content_type,
                 override_api_version=request_api_version,
             )
+            if proxy_path == "pipeline":
+                aiohttp_session = request.cookies.get("AIOHTTP_SESSION")
+                if not (sso_token := request.headers.get("X-BackendAI-SSO")):
+                    jwt_secret = request.app["config"]["pipeline"]["jwt"]["secret"]
+                    now = datetime.now(tz=timezone(timedelta(hours=9)))
+                    claims = {
+                        # Registered claims
+                        "exp": now + timedelta(hours=1),
+                        "iss": "Backend.AI Webserver",
+                        "iat": now,
+                        # Private claims
+                        "aiohttp_session": aiohttp_session,
+                        "access_key": api_session.config.access_key,
+                        # "secret_key": api_session.config.secret_key,
+                    }
+                    sso_token = jwt.encode(claims, key=jwt_secret, algorithm="HS256")
+                api_rqst.headers["X-BackendAI-SSO"] = sso_token
+                if session_id := (request_headers.get("X-BackendAI-SessionID") or aiohttp_session):
```

Comment:
```suggestion
                if session_id := request_headers.get("X-BackendAI-SessionID", aiohttp_session):
```
There is no such big deal but how about this?

Reply:
Plus, is it possible that `aiohttp_session` can be `None` here?

Reply:
1. This is to avoid unexpected results: `request_headers.get("X-BackendAI-SessionID", aiohttp_session)` will return `None` rather than `aiohttp_session` when `request_headers = {"X-BackendAI-SessionID": None}`.
2. If `aiohttp_session` is `None`, which means a session is not authenticated, then the pipeline service will invalidate and guide an user to log in.